src/main/webapp/script.js

```diff
@@ -0,0 +1,371 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+/**
+ * Function which is called when page loads.
+ * Add event listener to image upload button.
+ * Add canvases to the page, put sample image to them.
+ */
+document.addEventListener('DOMContentLoaded', () => {
+  const SAMPLE_IMAGE_URL = 'images/hadgehog.jpg';
+  const CANVAS_WIDTH = 300;
+
+  const uploadButton = document.getElementById('upload-image');
+  uploadButton.addEventListener('change', handleImageUpload);
+
+  const imagesContainer = document.getElementById('images-container');
+
+  // put canvases into image container on html page
+  const inputCanvas = createCanvas('input-canvas');
+  inputCanvas.setAttribute('width', CANVAS_WIDTH);
+  imagesContainer.append(inputCanvas);
+
+  const outputCanvas = createCanvas('output-canvas');
+  outputCanvas.setAttribute('width', CANVAS_WIDTH);
+  imagesContainer.append(outputCanvas);
+
+  // add sample image on page - original and blurred one.
+  blur(SAMPLE_IMAGE_URL);
+});
+
+/**
+ * Function which is called when user chooses file to upload.
+ * @param {Event} event Change event on image upload button.
+ */
+function handleImageUpload(event) {
+  if (!validateImageUpload()) {
+    return;
+  }
+
+  const imageUrl = URL.createObjectURL(event.target.files[0]);
+  blur(imageUrl);
+}
+
+/**
+ * Function to validate the uploaded image.
+ * @return {Boolean}
+ */
+function validateImageUpload() {
+  const fileName = document.getElementById('upload-image').value;
+
+  // if nothing was uploaded
+  if (fileName === '') {
+    alert('We can blur anything only if you upload an image');
+    return false;
+  }
+
+  // validate file extension
+  const SUPPORTED_EXTENSIONS = ['png', 'jpeg', 'jpg'];
+  const fileExtension = fileName.substring(
+      fileName.lastIndexOf('.') + 1).toLowerCase();
+
+  if (!SUPPORTED_EXTENSIONS.includes(fileExtension)) {
+    alert('We can support only jpg, jpeg and png images, sorry :(');
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Creates canvas with specified id.
+ * @param {string} id
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvas(id) {
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('id', id);
+
+  return canvas;
+}
+
+/**
+ * Creates canvas with width and height of an image.
+ * @param {Image} image
+ * @return {HTMLCanvasElement} canvas DOM element
+ */
+function createCanvasForImage(image) {
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('width', image.width);
+  canvas.setAttribute('height', image.height);
+
+  return canvas;
+}
+
+/**
+ * Function to handle all the blurring process
+ * and putting both blurred and original images on page.
+ * @param {string} imageUrl
+ */
+async function blur(imageUrl) {
```

Comment:
You can make the function name reflect its purpose. If the function not only blurs but also puts images on the page, you can call it 'blurAndPutImagesOnPage', or you can call it 'run' or 'perform' or something very generic, so that people know that it does not only blur but also does something else.